src/core/hle/service/news/news.cpp

```diff
@@ -2,18 +2,836 @@
+
+    return copy_size;
+}
+
+ResultVal<std::size_t> Module::GetNotificationMessage(const u32 notification_index, u8* message,
+                                                      const std::size_t size) {
+    if (!db.header.IsValid()) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    if (notification_index >= MAX_NOTIFICATIONS) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    const u32 notification_id = notification_ids[notification_index];
+
+    if (!db.notifications[notification_id].IsValid()) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    std::string message_file = fmt::format("/news{:03d}.txt", notification_id);
+
+    const auto result = LoadFileFromSavedata(message_file, size, message);
+
+    if (result.Failed()) {
+        return result.Code();
+    }
+
+    return result.Unwrap();
+}
+
+ResultVal<std::size_t> Module::GetNotificationImage(const u32 notification_index, u8* image,
+                                                    const std::size_t size) {
+    if (!db.header.IsValid()) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    if (notification_index >= MAX_NOTIFICATIONS) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    const u32 notification_id = notification_ids[notification_index];
+
+    if (!db.notifications[notification_id].IsValid()) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    std::string image_file = fmt::format("/news{:03d}.mpo", notification_id);
+
+    const auto result = LoadFileFromSavedata(image_file, size, image);
+
+    if (result.Failed()) {
+        return result.Code();
+    }
+
+    return result.Unwrap();
+}
+
+Result Module::SetNewsDBHeader(const NewsDBHeader* header, const std::size_t size) {
+    const std::size_t copy_size = std::min(sizeof(NewsDBHeader), static_cast<std::size_t>(size));
+
+    std::memcpy(&db.header, header, copy_size);
+
+    return SaveNewsDBSavedata();
+}
+
+Result Module::SetNotificationHeader(const u32 notification_index, const NotificationHeader* header,
+                                     const std::size_t size) {
+    if (notification_index >= MAX_NOTIFICATIONS) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    const u32 notification_id = notification_ids[notification_index];
+
+    const std::size_t copy_size = std::min(sizeof(NotificationHeader), size);
+
+    std::memcpy(&db.notifications[notification_id], header, copy_size);
+
+    return SaveNewsDBSavedata();
+}
+
+Result Module::SetNotificationHeaderOther(const u32 notification_index,
+                                          const NotificationHeader* header,
+                                          const std::size_t size) {
+    if (notification_index >= MAX_NOTIFICATIONS) {
+        return ERROR_INVALID_HEADER;
+    }
+
+    const u32 notification_id = notification_ids[notification_index];
+
+    const std::size_t copy_size = std::min(sizeof(NotificationHeader), size);
+
+    std::memcpy(&db.notifications[notification_id], header, copy_size);
+
+    return ResultSuccess;
```

Comment:
```suggestion
    const u32 notification_id = notification_ids[notification_index];
    const std::size_t copy_size = std::min(sizeof(NotificationHeader), size);
    std::memcpy(&db.notifications[notification_id], header, copy_size);
    return ResultSuccess;
```
Can be seen as a nitpick but the spacing can be reduced here which will make the function more compact and less overwhelming to read. ditto for similar cases above.